src/Symfony/Component/Security/Http/RememberMe/RememberMeDetails.php

```diff
@@ -142,4 +135,44 @@ public function toString(): string
         // $userIdentifier is encoded because it might contain COOKIE_DELIMITER, we assume other values don't
         return implode(self::COOKIE_DELIMITER, [strtr($this->userFqcn ?? '', '\\', '.'), strtr(base64_encode($this->userIdentifier), '+/=', '-_~'), $this->expires, $this->value]);
     }
+
+    private static function collectConstructorArguments(string $userFqcn, string $userIdentifier, int $expires, string $value): array
+    {
+        $constructor = new \ReflectionMethod(static::class, '__construct');
+
+        if (self::class === $constructor->class) {
+            return [
+                $userFqcn,
+                $userIdentifier,
+                $expires,
+                $value,
+                false,
+            ];
+        }
+
+        $requiredArguments = 0;
+
+        foreach ($constructor->getParameters() as $parameter) {
+            if (!$parameter->isOptional()) {
+                ++$requiredArguments;
+            }
+        }
+
+        if (4 === $requiredArguments) {
```

Comment:
just in case:
```suggestion
        if (3 < $requiredArguments) {
```

Reply:
works for me, even though having more than four required arguments will lead to a PHP error afterwards anyway